client/state/site-settings/exporter/actions.js

```diff
@@ -26,41 +43,168 @@ export function setPostType( postType ) {
 }
 
 /**
- * Sends a request to the server to start an export.
+ * Sets one of the advanced export settings values in the UI
  *
+ * @param  {string} section   The name of the section containing the setting - 'posts', 'pages', or 'feedback'
+ * @param  {string} setting   The name of the setting
+ * @param  {any}    value     The new value for the setting
+ * @return {Object}           Action object
+ */
+export function setAdvancedSetting( section, setting, value ) {
+	return {
+		type: SET_EXPORTER_ADVANCED_SETTING,
+		section,
+		setting,
+		value
+	};
+}
+
+/**
+ * Ensure that we are fetching or have fetched the available export
+ * settings for the current site.
+ *
+ * @param  {int}      siteId  The ID of the site for which to retrieve export settings
  * @return {Function}         Action thunk
  */
-export function startExport() {
+export function ensureHasSettings() {
+	return ( dispatch, getState ) => {
+		const state = getState();
+		const selectedSite = getSelectedSite( state );
+		const currentDataSiteId = getDataState( state ).forSiteId;
+
+		if ( selectedSite ) {
+			if ( selectedSite.ID !== currentDataSiteId ) {
+				dispatch( fetchExportSettings( selectedSite.ID ) );
+			}
+		}
```

Comment:
*lodash*'s `get()` could help us here to eliminate a layer of nesting. it won't throw an error if the values are undefined (and additionally can provide a default value if the one we want doesn't exist)

```js
import get from 'lodash/object/get';

if ( get( selectedSite, 'ID' ) !== currentDataSiteId ) {
	…
}
```